app/components/profile-nav.js

```diff
@@ -61,4 +61,21 @@ export default Component.extend({
     return !this.features.get('enterpriseVersion') && !!billingEndpoint;
   }),
 
+  didRender() {
+    const isUser = this.model.isUser;
+    const login = this.model.login;
+    const allowance = this.model.allowance;
+    const plansPath = isUser ? `https://travis-ci.com/account/${config.planSuffix}` : `https://travis-ci.com/organizations/${login}/${config.planSuffix}`;
+    const settingsPath = isUser ? `https://travis-ci.com/account/${config.settingsSuffix}` : `https://travis-ci.com/organizations/${login}/${config.settingsSuffix}`;
+
+    if (allowance && !allowance.public_repos) {
+      this.flashes.warning(`Builds have been temporarily disabled for public repositories due to a negative credit balance. \
+                            Please go to the <a href="${plansPath}">Plan page</a> to replenish your credit balance or alter your \
+                            <a href="${settingsPath}">OSS Credits consumption setting</a>`);
+    } else if (allowance && !allowance.private_repos) {
+      this.flashes.warning(`Builds have been temporarily disabled for private repositories due to a negative credit balance. \
+                            Please go to the <a href="${plansPath}">Plan page</a> to replenish your credit balance`);
```

Comment:
These are huge HTML blobs being passed to a service method, which doesn't look right to me. I'd recommend you to implement these messages as templates and then display by the service. For example, just an idea - create a new type of messages: `custom` (similar to `warning` that you've created), which would receive a template name as a parameter along with the context object that provides the data required to properly render the template.